### PR TITLE
Add security context to kafka and CC pod and containers

### LINKED ISSUE #138: Implement securityContext
**Is your feature request related to a problem? Please describe.**
In some cases, it would be fine if deployed pods (kafka, cruise-control) `securityContext` can be defined.

**Describe the solution you'd like to see**
The `securityContext` should be defined in `KafkaCluster` CR .

### PR DESCRIPTION
**Pull request by baluchicken**
| Q               | A
| --------------- | ---
| Bug fix?        | no|
| New feature?    |yes|
| API breaks?     | no|
| Deprecations?   | no|
| Related tickets | fixes #138 
| License         | Apache 2.0

### What's in this PR?

This PR adds support for podSecurityContext and container securityContext to the operator.

### Checklist

- [x] Implementation tested
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/developer-guide/blob/master/docs/coding-style/error-handling-guide.md)
- [x] Logging code meets the guideline
- [x] User guide and development docs updated (if needed)

### CODE REVIEW

**Review by bonifaido — DISMISSED** [review]

File: pkg/sdk/v1beta1/kafkacluster_types.go [hunk 1]
```diff
@@ -172,7 +175,10 @@ type CruiseControlConfig struct {
 	// Volumes define some extra Kubernetes Volumes for the CruiseControl Pods.
 	Volumes []corev1.Volume `json:"volumes,omitempty"`
 	// VolumeMounts define some extra Kubernetes Volume mounts for the CruiseControl Pods.
-	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty"`
+	VolumeMounts       []corev1.VolumeMount       `json:"volumeMounts,omitempty"`
+	PodSecurityContext *corev1.PodSecurityContext `json:"podSecurityContext,omitempty"`
+	// SecurityContext allow to set security context for the cruise-control container
```
Comment (hi-im-aren) [inline pkg/sdk/v1beta1/kafkacluster_types.go:180]:
```suggestion
	// SecurityContext allows to set security context for the CruiseControl container
```

File: pkg/sdk/v1beta1/kafkacluster_types.go [hunk 2]
```diff
@@ -136,7 +136,10 @@ type BrokerConfig struct {
 	// and the operator supposes that the user is aware of how scheduling is done by kubernetes
 	// Affinity could be set through brokerConfigGroups definitions and can be set for individual brokers as well
 	// where letter setting will override the group setting
-	Affinity *corev1.Affinity `json:"affinity,omitempty" protobuf:"bytes,18,opt,name=affinity"`
+	Affinity           *corev1.Affinity           `json:"affinity,omitempty" protobuf:"bytes,18,opt,name=affinity"`
+	PodSecurityContext *corev1.PodSecurityContext `json:"podSecurityContext,omitempty"`
+	// SecurityContext allow to set security context for the kafka container
```
Comment (hi-im-aren) [inline pkg/sdk/v1beta1/kafkacluster_types.go:141]:
```suggestion
	// SecurityContext allows to set security context for the kafka container
```

**Review by hi-im-aren — APPROVED** [review]

**Review by pbalogh-sa — APPROVED** [review]

**Review by pbalogh-sa — DISMISSED** [review]

File: pkg/sdk/v1beta1/kafkacluster_types.go [hunk 3]
```diff
@@ -136,7 +136,10 @@ type BrokerConfig struct {
 	// and the operator supposes that the user is aware of how scheduling is done by kubernetes
 	// Affinity could be set through brokerConfigGroups definitions and can be set for individual brokers as well
 	// where letter setting will override the group setting
-	Affinity *corev1.Affinity `json:"affinity,omitempty" protobuf:"bytes,18,opt,name=affinity"`
+	Affinity           *corev1.Affinity           `json:"affinity,omitempty" protobuf:"bytes,18,opt,name=affinity"`
```
Comment (stoader) [inline pkg/sdk/v1beta1/kafkacluster_types.go:139]: The  `protobuf:"bytes,18` seems to be a leftover from a previous PR. I'd remove it as it's not used.

**Review by stoader — CHANGES_REQUESTED** [review]

**Review by adamantal — COMMENTED** [review]
LGTM. 
Could you please also add a test that configures a non nil `SecurityContext` in the spec and assert that it is correctly set in the created pod?

**Review by stoader — APPROVED** [review]

**Review by pbalogh-sa — APPROVED** [review]

**Review by adamantal — APPROVED** [review]
